2024-july/demo-3/hospital-service/hospital_service.bal

```diff
@@ -0,0 +1,168 @@
+import ballerina/http;
+import ballerina/uuid;
+
+listener http:Listener hospitalEP = new (9090);
+
+service /hospital on hospitalEP {
+    # Get a list of patients
+    #
+    # + return - A list of patients 
+    isolated resource function get patients() returns Patient[] {
+        lock {
+            return patientTable.toArray().clone();
+        }
+    }
+
+    # Get a patient by ID
+    #
+    # + return - returns can be any of following types 
+    # http:Ok (Patient details)
+    # http:NotFound (Patient not found)
+    isolated resource function get patients/[string patientId]() returns Patient|http:NotFound {
+        lock {
+            if !patientTable.hasKey(patientId) {
+                return {body: "Patient not found"};
+            }
+            return patientTable.get(patientId).clone();
```

Comment:
Why do we need clone here?

Reply:
We need it here to ensure that the Patient record that we retrieve is not mutated, therefore isolated property is ensured. I had to add to make it thread-safe, otherwise a lot of concurrency warnings are getting printed, and in the demo it doesn't look nice.